game-core/src/main/java/games/strategy/engine/data/AttachmentProperty.java

```diff
@@ -0,0 +1,84 @@
+package games.strategy.engine.data;
+
+import java.util.function.Supplier;
+
+public interface AttachmentProperty<T> {
+
+  void setValue(String string) throws GameParseException;
+
+  void setValue(T value) throws GameParseException;
+
+  T getValue();
+
+  void resetValue();
+
+  static <T> AttachmentProperty<T> of(
+      final GameParsingConsumer<T> setter,
+      final GameParsingConsumer<String> stringSetter,
+      final Supplier<T> getter,
+      final Runnable resetter) {
+    return new AttachmentProperty<T>() {
+
+      @Override
+      public void setValue(String string) throws GameParseException {
+        stringSetter.accept(string);
+      }
+
+      @Override
+      public void setValue(T value) throws GameParseException {
+        setter.accept(value);
+      }
+
+      @Override
+      public T getValue() {
+        return getter.get();
+      }
+
+      @Override
+      public void resetValue() {
+        resetter.run();
+      }
+    };
+  }
+  
+  static <T> AttachmentProperty<T> of(
+      final GameParsingConsumer<T> setter,
+      final GameParsingConsumer<String> stringSetter,
+      final Supplier<T> getter) {
+    return of(setter, stringSetter, getter, () -> throwIllegalStateException("No Resetter"));
+  }
+
+  static AttachmentProperty<String> of(
+      final GameParsingConsumer<String> setter,
+      final Supplier<String> getter,
+      final Runnable resetter) {
+    return of(setter, setter, getter, resetter);
+  }
+
+  static <T> AttachmentProperty<T> of(final Supplier<T> getter) {
+    return of(
+        t -> throwIllegalStateException("No Setter"),

```

Comment:
I think for most (if not all) of these cases, `UnsupportedOperationException` would be more appropriate than `IllegalStateException`.

Reply:
@RoiEXLab Bumping this.  What do you think?